lxd/device/device_utils_network.go

```diff
@@ -101,9 +101,13 @@ func networkRemoveInterfaceIfNeeded(state *state.State, nic string, current inst
 }
 
 // networkCreateVlanDeviceIfNeeded creates a VLAN device if doesn't already exist.
-func networkCreateVlanDeviceIfNeeded(state *state.State, parent string, vlanDevice string, vlanID string) (string, error) {
+func networkCreateVlanDeviceIfNeeded(state *state.State, parent string, vlanDevice string, vlanID string, gvrp string) (string, error) {
 	if vlanID != "" {
-		created, err := network.VLANInterfaceCreate(parent, vlanDevice, vlanID)
+
+		if gvrp != "" {
```

Comment:
@TimRots I'd be tempted to make the gvrp argument a bool (you can then pass that in via `shared.IsTrue(d.config["gvrp"])` and then pass that onto `VLANInterfaceCreate` as a bool.